### PR TITLE
ci: update to node v18

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1166]

Use Node `v18.16.0` since i get `TypeError: The "emitter" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal` with `npm run test` due to Node upstream bug with Node's unidici library (Node v18.15.0 <=)

## Test Plan
- Check if all CI are green
- Check if webui works on local

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1166]: https://hpe-aiatscale.atlassian.net/browse/WEB-1166?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by keita-determined** [comment]
@ashtonG do you know why [react-unit-test CI](https://app.circleci.com/pipelines/github/determined-ai/determined/38304/workflows/013402b3-75fa-4dc8-9c18-5f468f380a7f/jobs/1406817) is hanging forever? 
It finished fine on local, but `make -C webui/react test-ci` doesnt stop on CI even after all test cases are finished.

image

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ git clone --recurse-submodules https://github.com/determined-ai/determined.git
   - python3-venv
   - python3-wheel
   - python3-dev
-- Node (>= 16.13, < 20)
+- Node (>= 18.16.0, < 19)
```
Comment (ashtonG) [inline CONTRIBUTING.md:54]: shouldn't we keep the max version to 20?

Comment (keita-determined) [reply CONTRIBUTING.md:54]: i dont think its a good idea because node v19 will be EOL in 3 weeks https://endoflife.date/nodejs

Comment (keita-determined) [reply CONTRIBUTING.md:54]: and the max version was up v19, but i think we can support v18 and v20?

Comment (ashtonG) [reply CONTRIBUTING.md:54]: Ah, I see.

**Review by ashtonG — APPROVED** [review]

**Pull request merged**